cloud/services/domains.go

```diff
@@ -199,11 +167,10 @@ func DeleteDomainRecord(ctx context.Context, mscope *scope.MachineScope, hostnam
 	}
 
 	// If record is A type, verify ownership
-	if recordType != "TXT" {
-		txtRecordValueString := createSHA256HashOfString(mscope.LinodeMachine.Name)
-		isOwner, ownerErr := IsDomainRecordOwner(ctx, mscope, hostname, "owner:"+txtRecordValueString, domainID)
-		if ownerErr != nil {
-			return ownerErr
+	if dnsEntry.DNSRecordType != "TXT" {
```

Comment:
```suggestion
	if dnsEntry.DNSRecordType != linodego.RecordTypeTXT {
```
Although I wonder if in line with the comment we should be checking only A records

Reply:
done and updated the comment